### PR TITLE
Move licenses JSON file to root

### PR DESCRIPTION
**Pull request by estroz**
`*`: moved `all_licenses.json` from `/installer/scripts/license-gen/licenses/` -> `/`.
`installer/scripts/license-gen`: refactored `collect_project_licenses.sh` to write `all_licenses.json` to `/`.

### CODE REVIEW

File: installer/scripts/license-gen/collect_project_licenses.sh [hunk 1]
```diff
@@ -26,29 +26,30 @@
 
 set -eu
 
+# Set relative environment
 DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
-FINAL_LICENSE_DIR="$DIR"/licenses
-FINAL_LICENSE_FILE="$FINAL_LICENSE_DIR"/all_licenses.json
+ROOT="$DIR/../../.."
+FINAL_LICENSE_FILE="$ROOT"/all_licenses.json
```
Comment (robszumski) [inline installer/scripts/license-gen/collect_project_licenses.sh]: can we rename this to `bill-of-materials.json` for consistency?

**Comment by sym3tri** [comment]
@spoonben we're ready to go once this merges.

**Comment by spoonben** [comment]
@sym3tri what's up with these? (and some others that follow) https://github.com/estroz/tectonic-installer/blob/bcf750237e23c06b5f0deb506ba88d39b50f18eb/all_licenses.json#L537-L651

**Comment by sym3tri** [comment]
@spoonben ignore those for now. They'll be removed soon.

**Pull request merged**